dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -48,12 +48,23 @@ def element_stale?(element)
   true
 end
 
-def page_load(wait_until_unload)
-  if wait_until_unload
-    html = @browser.find_element(tag_name: 'html')
+def page_load(wait = true, blank_tab: false)
+  if wait
+    root = @browser.find_element(css: ':root')
+    tabs = @browser.window_handles if wait == 'tab'
     yield
-    wait_until {element_stale?(html)}
-    navigate_to(@browser.current_url)
+    if tabs
+      new_tab = wait_until {(@browser.window_handles - tabs).first}
+      @browser.switch_to.window(new_tab)
+    end
+    wait_until {element_stale?(root)}
+    unless blank_tab
+      wait_until do
+        (url = @browser.current_url) != '' &&
+           url != 'about:blank' &&
+          @browser.execute_script('return document.readyState;') == 'complete'
```

Comment:
This extra `wait_until` addresses some flaky failures I saw where a newly-created tab will sometimes temporarily show 'about:blank' document before loading its destination URL, so we want to wait until the destination page shows up. (Except for the case where we _want_ to load a blank tab as our step.)